main.go

```diff
@@ -21,70 +15,20 @@ var (
 	printVersion bool
 )
 
-var goneBranch = regexp.MustCompile(`(?m)^(?:\*| ) ([^\s]+)\s+[a-z0-9]+ \[[^:\n]+: gone\].*$`)
-
-func checkForError(e error) {
-	if e != nil {
-		fmt.Fprintln(os.Stderr, e)
-		os.Exit(1)
-	}
-}
-
 func main() {
 	flag.Parse()
-
-	if printVersion {
-		fmt.Println(version)
-		return
-	}
-
 	directory := flag.Arg(0) // first trailing argument after flags (if any)
-	if directory == "" {
-		directory = "."
-	}
-
-	git := gUtils.NewExecutor(directory)
 
-	goneBranches := []string{}
-	branchesToDelete := []string{}
+	intent := commands.NewIntent(printVersion)
 
-	git.Fetch().Prune().ListRemoteBranches()
-	submatches := goneBranch.FindAllStringSubmatch(git.Output, -1)
-	for _, matches := range submatches {
-		if len(matches) == 2 && matches[1] != "" {
-			if interactive {
-				goneBranches = append(goneBranches, matches[1])
-			} else {
-				branchesToDelete = append(branchesToDelete, matches[1])
-			}
-		}
-	}
-
-	prompt := &survey.MultiSelect{
-		Message: "Use the spacebar to select the branches you would like to delete:",
-		Options: goneBranches,
-	}
-	survey.AskOne(prompt, &branchesToDelete, nil)
-
-	for _, branch := range branchesToDelete {
-		git.Delete(branch, force)
-	}
-
-	if len(git.DeletedBranches) > 0 {
-		fmt.Println("Deleted branches:")
-		for _, branch := range git.DeletedBranches {
-			fmt.Println(branch)
-		}
-	}
-
-	if len(git.BranchDeletionErrors) > 0 {
-		fmt.Println("Errors:")
-		for _, err := range git.BranchDeletionErrors {
-			fmt.Printf("[%s]: %s", err.Branch, err.Msg)
-		}
+	switch i := intent.Is(); i {
+	case commands.VersionIntent:
+		commands.Version(version)
+		return
+	default:
```

Comment:
Any reason you don't want to be explicit with `commands. DefaultIntent`?

Reply:
Not necessarily. This was a bit more terse, and would still make sure the app did *something* even if the `intent` was screwed up. 

Do you think it's wise to be more explicit?

Reply:
I might have opted to include a case for `commands.DefaultIntent`, but keep the `default:` and print something/fail/etc.